### DISCUSSION TITLE
\meson.build:1:0: ERROR: Unknown compiler(s):

### DISCUSSION THREAD

**Discussion by QFirebal** [snipem/gt7dashboard#19] [Q&A]
Hi everyone,

i am pretty new when it is about github and especially python. I havedownlaoded the zip, installed python and entered my ip in the bat file i recive following error:
error: subprocess-exited-with-error

  × Preparing metadata (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [21 lines of output]
      + meson setup C:\Users\chris\AppData\Local\Temp\pip-install-89reaesx\scipy_4d79aadf1460444aaefa918f6ce175b3 C:\Users\chris\AppData\Local\Temp\pip-install-89reaesx\scipy_4d79aadf1460444aaefa918f6ce175b3\.mesonpy-ll9jygqt -Dbuildtype=release -Db_ndebug=if-release -Db_vscrt=md --native-file=C:\Users\chris\AppData\Local\Temp\pip-install-89reaesx\scipy_4d79aadf1460444aaefa918f6ce175b3\.mesonpy-ll9jygqt\meson-python-native-file.ini
      The Meson build system
      Version: 1.2.2
      Source dir: C:\Users\chris\AppData\Local\Temp\pip-install-89reaesx\scipy_4d79aadf1460444aaefa918f6ce175b3
      Build dir: C:\Users\chris\AppData\Local\Temp\pip-install-89reaesx\scipy_4d79aadf1460444aaefa918f6ce175b3\.mesonpy-ll9jygqt
      Build type: native build
      Project name: SciPy
      Project version: 1.9.3
      WARNING: Failed to activate VS environment: Could not find C:\Program Files (x86)\Microsoft Visual Studio\Installer\vswhere.exe

      ..\meson.build:1:0: ERROR: Unknown compiler(s): [['icl'], ['cl'], ['cc'], ['gcc'], ['clang'], ['clang-cl'], ['pgcc']]
      The following exception(s) were encountered:
      Running `icl ""` gave "[WinError 2] Das System kann die angegebene Datei nicht finden"
      Running `cl /?` gave "[WinError 2] Das System kann die angegebene Datei nicht finden"
      Running `cc --version` gave "[WinError 2] Das System kann die angegebene Datei nicht finden"
      Running `gcc --version` gave "[WinError 2] Das System kann die angegebene Datei nicht finden"
      Running `clang --version` gave "[WinError 2] Das System kann die angegebene Datei nicht finden"
      Running `clang-cl /?` gave "[WinError 2] Das System kann die angegebene Datei nicht finden"
      Running `pgcc --version` gave "[WinError 2] Das System kann die angegebene Datei nicht finden"

      A full log can be found at C:\Users\chris\AppData\Local\Temp\pip-install-89reaesx\scipy_4d79aadf1460444aaefa918f6ce175b3\.mesonpy-ll9jygqt\meson-logs\meson-log.txt
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.

i am on Windows. Can anyone help?

**Comment by snipem** [comment]
You don’t have to build it. Have a look at this section of the README: https://github.com/snipem/gt7dashboard#how-to-run

**Reply by QFirebal** [reply]
Thanks i am a s step further, i can start it now but when i open the Bokeh app in browser its stays empty and i recive following logs:

Windows PowerShell
Copyright (C) Microsoft Corporation. Alle Rechte vorbehalten.

Installieren Sie die neueste PowerShell für neue Funktionen und Verbesserungen! https://aka.ms/PSWindows

PS C:\Users\chris\Desktop\gt7dashboard-1.1> set GT7_PLAYSTATION_IP=XXX
PS C:\Users\chris\Desktop\gt7dashboard-1.1> python -m bokeh serve .
2023-10-17 10:25:39,348 Starting Bokeh server version 3.3.0 (running on Tornado 6.3.3)
2023-10-17 10:25:39,349 User authentication hooks NOT provided (default user enabled)
2023-10-17 10:25:39,353 Bokeh app running at: http://localhost:5006/gt7dashboard-1.1
2023-10-17 10:25:39,353 Starting Bokeh server with process id: 11460
2023-10-17 10:25:55,273 Error running application handler <bokeh.application.handlers.directory.DirectoryHandler object at 0x0000020007056030>: No module named 'salsa20'
File 'gt7communication.py', line 12, in <module>:
from salsa20 import Salsa20_xor Traceback (most recent call last):
  File "C:\Users\chris\AppData\Local\Programs\Python\Python312\Lib\site-packages\bokeh\application\handlers\code_runner.py", line 229, in run
    exec(self._code, module.__dict__)
  File "C:\Users\chris\Desktop\gt7dashboard-1.1\main.py", line 24, in <module>
    from gt7dashboard import gt7communication, gt7diagrams, gt7help, gt7helper, gt7lap
  File "C:\Users\chris\Desktop\gt7dashboard-1.1\gt7dashboard\gt7communication.py", line 12, in <module>
    from salsa20 import Salsa20_xor
ModuleNotFoundError: No module named 'salsa20'

2023-10-17 10:25:55,541 WebSocket connection opened
2023-10-17 10:25:55,542 ServerConnection created

**Reply by snipem** [reply]
Have a look here: https://github.com/snipem/gt7dashboard/discussions/9#discussioncomment-6361183
